pkg/osvscanner/osvscanner.go

```diff
@@ -357,29 +357,53 @@ func scanDebianDocker(r *output.Reporter, query *osv.BatchedQuery, dockerImageNa
 	return nil
 }
 
-// Filters response according to config, returns number of responses removed
-func filterResponse(r *output.Reporter, query osv.BatchedQuery, resp *osv.BatchedResponse, configManager *config.ConfigManager) int {
+// Filters vulnerabilities and groups according to config, returns filtered vulnerabilities and groups
+func filterVulns(r *output.Reporter, configToUse config.Config, vulns []models.Vulnerability, groups []models.GroupInfo) ([]models.Vulnerability, []models.GroupInfo) {
 	hiddenVulns := map[string]config.IgnoreEntry{}
+	outGroups := []models.GroupInfo{}
+	outVulns := []models.Vulnerability{}
 
-	for i, result := range resp.Results {
-		var filteredVulns []osv.MinimalVulnerability
-		configToUse := configManager.Get(r, query.Queries[i].Source.Path)
-		for _, vuln := range result.Vulns {
-			ignore, ignoreLine := configToUse.ShouldIgnore(vuln.ID)
+	for _, group := range groups {
+		keepIDs := []string{}
+		for i, id := range group.IDs {
+			ignore, ignoreLine := configToUse.ShouldIgnore(id)
 			if ignore {
-				hiddenVulns[vuln.ID] = ignoreLine
+				hiddenVulns[id] = ignoreLine
+				if ignoreLine.ShouldIncludeAliases() {
+					for _, id := range keepIDs {
+						hiddenVulns[id] = ignoreLine
+					}
+					for _, id := range group.IDs[i+1:] {
+						hiddenVulns[id] = ignoreLine
+					}
+					keepIDs = []string{}
+
```

Comment:
I think this can just be simplified to
```suggestion
					for _, id := range group.IDs {
						hiddenVulns[id] = ignoreLine
					}
					keepIDs = []string{}
```

Reply:
Simplified a bit when taking out `includeAliases`